src/coreComponents/meshUtilities/CPMesh/CPMeshBuilder.hpp

```diff
@@ -0,0 +1,262 @@
+
+
+private:
+
+  /**
+   * @brief Once the file is read, process the Eclipse keywords and fill the datastructure describing the conforming mesh
+   * @details The post-processing proceeds in the following steps:
+   *   1) For each corner-point cell, compute the position of the eight corner-point vertices
+   *   2) Filter out duplicate vertices
+   *   3) Construct faces (no implemented yet)
+   */
+  void postProcessMesh();
+
+  /**
+   * @brief Loop over corner-point cells to compute the position of the eight corner-point vertices in each cell
+   */
+  void buildCornerPointCells();
+
+  /**
+   * @brief Using ZCORN and COORD, compute the position of the eight corner-point vertices in cell (i,j,k)
+   * @param i index in the x-direction
+   * @param j index in the y-direction
+   * @param k index in the z-direction
+   * @param nXLocal number of cells in the x-direction on my rank
+   * @param nYLocal number of cells in the y-direction on my rank
+   * @param coord local content of the COORD keyword
+   * @param zcorn local content of the ZCORN keyword
+   * @param xPos x-position of the eight corner-point vertices
+   * @param yPos y-position of the eight corner-point vertices
+   * @param zPos z-position of the eight corner-point vertices
+   */
+  bool processActiveHexahedron( localIndex const i,
+                                localIndex const j,
+                                localIndex const k,
+                                localIndex const nXLocal,
+                                localIndex const nYLocal,
+                                localIndex const iMinOverlap,
+                                localIndex const iMaxOverlap,
+                                localIndex const jMinOverlap,
+                                localIndex const jMaxOverlap,
+                                array1d< real64 > const & coord,
+                                array1d< real64 > const & zcorn,
+                                array1d< real64 > & xPos,
+                                array1d< real64 > & yPos,
+                                array1d< real64 > & zPos,
+                                array1d< bool > & cPVertexIsInside ) const;
+
+  /**
+   * @brief Loop over the corner-point vertices and filter out duplicates.
+   */
+  void filterVertices();
+
+  /**
+   * @brief For the non-conforming case, build faces (not implemented yet)
+   */
+  void buildFaces();
+
+
+  /// Object holding the mesh (and MPI partition) dimensions
+  CPMeshDimensions m_meshDims;
+
+  /// Object storing vertex data
+  CPMeshVertices m_meshVertices;
+
+  /// Object storing face data
+  CPMeshFaces m_meshFaces;
+
+  /// Object storing cell data
+  CPMeshCells m_meshCells;
+
+  /// Object in charge of parsing the GRDECL file
+  CPMeshParser m_meshParser;
+
+  /// Object holding the communication info
+  CPMeshComm m_meshComm;
+
+  /// Name of the mesh
+  string m_meshName;
```

Comment:
Maybe remove all the `mesh` from the names?

Reply:
Yes, I removed `mesh` from all these names